### PR TITLE
Sign in sign out caching

### PR DESCRIPTION
**Pull request by evguenirousselot**
In this pull request I finalized the sign in/sign out caching functionality started during the previous sprint.

1) Pressing the sign out button will erase the caching and will require a new login with google

2) It is now possible to sign in using your username and the corresponding data will be fetched. As you've previously signed in using google, we assume that you're the owner of this email address and the pair (email,current username) will log you in 

Full tests are also added

Refactoring of the code may later come

This closes issue #60 #69

### CODE REVIEW

File: app/src/androidTest/java/com/github/orkest/View/auth/SignUpFormTest.kt [hunk 1]
```diff
@@ -130,4 +136,25 @@ class SignUpFormTest {
         intended((hasComponent(MainActivity::class.java.name)))
         Intents.release()
     }
+
+    @Test
```
Comment (yasmineben510) [inline app/src/androidTest/java/com/github/orkest/View/auth/SignUpFormTest.kt:140]: Shouldn't these added tests be in the SignInTest not the SignUpFormTest file?

File: app/src/main/java/com/github/orkest/View/auth/SignUpForm.kt [hunk 2]
```diff
@@ -154,6 +157,51 @@ fun SignUpForm(navController: NavController, viewModel: AuthViewModel) {
                         )
                     }
                 }
+
```
Comment (yasmineben510) [inline app/src/main/java/com/github/orkest/View/auth/SignUpForm.kt:160]: I don't think this part should be in the same UI as the sign Up page, we should keep the UIs for sign up and for sign in separated

**Review by yasmineben510 — CHANGES_REQUESTED** [review]
Your code is well writtent but with regard to the functionality, I think the goal was more that if a user is already registered in the database, he should have to go through the sign in button once (not once in the sign in page and then again in the create profile)

Ideally, the goal would be that its e-mail is retrieved from the database to find the corresponding username to directly launch the mainActivity without going through the sign up page(which means we also need to store the e-mail at signup which is possible since the authViewModel is shared between the signUp and signIn pages).

File: app/src/androidTest/java/com/github/orkest/View/auth/SignUpFormTest.kt (hunk 1 above)
Comment (evguenirousselot) [reply app/src/androidTest/java/com/github/orkest/View/auth/SignUpFormTest.kt:140]: Yes you're right, I'll move it

File: app/src/main/java/com/github/orkest/View/auth/SignUpForm.kt (hunk 2 above)
Comment (evguenirousselot) [reply app/src/main/java/com/github/orkest/View/auth/SignUpForm.kt]: Yes you're right, I've changed it

**Review by yasmineben510 — APPROVED** [review]
Thank you for implementing the requested changes ! The new code looks good to me! Thank you for your work!

**Pull request merged**